src/ros_env.hpp

```diff
@@ -89,7 +89,8 @@ static void setMasterURI( const std::string& uri, const std::string& network_int
   remap["__master"] = uri;
   remap["__ip"] = ::naoqi::ros_env::getROSIP(network_interface);
   // init ros without a sigint-handler in order to shutdown correctly by naoqi
-  ros::init( remap, ::naoqi::ros_env::getPrefix().empty()?std::string("naoqi_driver_node"):(::naoqi::ros_env::getPrefix()) , ros::init_options::NoSigintHandler );
+  const char* ns_env = std::getenv("ROS_NAMESPACE");
+  ros::init( remap, (ns_env==NULL)?(std::string("naoqi_driver_node")):(::naoqi::ros_env::getPrefix()) , ros::init_options::NoSigintHandler );
```

Comment:
So what happens if I do not define a ROS_NAMESPACE variable and do not define a prefix either ? Isn't it the same as before (no namespace found) ?